pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java

```diff
@@ -78,6 +97,19 @@ protected void initChannel(SocketChannel ch) throws Exception {
         // ServerCnx ends up reading higher number of messages and broker can not throttle the messages by disabling
         // auto-read.
         ch.pipeline().addLast("flowController", new FlowControlHandler());
-        ch.pipeline().addLast("handler", new ServerCnx(pulsar));
+        ServerCnx cnx = new ServerCnx(pulsar);
+        ch.pipeline().addLast("handler", cnx);
+
+        connections.put(ch.remoteAddress(), cnx);
+    }
+
+    private void refreshAuthenticationCredentials() {
+        connections.asMap().values().forEach(cnx -> {
+            try {
+                cnx.refreshAuthenticationCredentials();
+            } catch (Throwable t) {
+                log.warn("[{}] Failed to refresh auth credentials", cnx.getRemoteAddress());
```

Comment:
Since you're using Cache to hold the connections, and the lifetime of the entry in connections is based on whether it's been GC'd, this will likely spam the logs. Maybe only log if after throwing, the cnx is still in connected state.

Reply:
When we get the ref to `cnx`, it will be a strong ref to it, so we're not going to loose it during the iteration. Then, inside that method, i'm checking for 

```java
   if (getState() != State.Connected || !isActive) {
            // Connection is either still being established or already closed.
            return;
        }
```

to avoid getting errors on a connections that's already gone